src/main/java/edu/njit/jerse/ashe/utils/JavaCodeCorrector.java

```diff
@@ -247,6 +264,24 @@ public String checkedFileError(String targetFile) {
     }
 
     /**
+	 * Returns the root path for a given Java file, using a method signature to resolve package structure.
+	 * 
+	 * @param targetFile	the path to a Java file
+	 * @param targetMethod	method signature present in the file
+	 * @return	root path based on targetFile path
+	 */
+	private static String getRootPath(String targetFile, String targetMethod) {
+		Path rootPath = Path.of(targetFile);
+	    long targetMethodDots = targetMethod.chars().filter(x -> x == '.').count();
+	    for(long i = 0; i < targetMethodDots + 1; i++) {
+	    	rootPath = rootPath.getParent();
+	    	if(rootPath == null)
```

Comment:
As a matter of style, avoid single-line `if` statements without braces in Java. They're a maintenance nightmare. For an example, see https://medium.com/@jonathanabrams/single-line-if-statements-2565c62ff492